### PR TITLE
Changing analytics event

### PR DESCRIPTION
**Pull request by ksebina**
Created a separate event for the 3P button click and now updating code to log the correct event, representing one of the flows for b/232215758.
This change deals with immediate redirect following the button click, with a delay of very short/unperceivable time using setTimeout.

### CODE REVIEW

**Review by fikus — CHANGES_REQUESTED** [review]
Can you add a test exercising the change?

**Review by jpettitt — COMMENTED** [review]
Was anybody external already using this event under the old name?

**Review by jpettitt — APPROVED** [review]

File: src/utils/gaa-test.js [hunk 1]
```diff
@@ -829,6 +829,29 @@ describes.realWin('GaaMeteringRegwall', {}, () => {
...
+    it('check 3p button click', () => {
+      GaaMeteringRegwall.showWithNative3PRegistrationButton({
+        authorizationUrl: GOOGLE_3P_AUTH_URL,
+        caslUrl: CASL_URL,
+      });
+
+      // Click button.
+      self.document.getElementById(GOOGLE_3P_SIGN_IN_BUTTON_ID).click();
+
+      sandbox.stub(self, 'setTimeout').callsFake((callback, timeout) => {
```
Comment (hartmannr76) [inline src/utils/gaa-test.js:842]: I think this needs to be setup before you click the button

**Review by hartmannr76 — APPROVED** [review]
LGTM. Maybe we should also make a tracking ticket to fix the full downstream calls to be chained. Lots of unmanaged promises

File: src/utils/gaa.js [hunk 2]
```diff
@@ -971,11 +974,13 @@ export class GaaMeteringRegwall {
     buttonEl.addEventListener('click', () => {
       // Track button clicks.
       logEvent({
-        analyticsEvent: AnalyticsEvent.ACTION_SHOWCASE_REGWALL_GSI_CLICK,
+        analyticsEvent: AnalyticsEvent.ACTION_SHOWCASE_REGWALL_3P_BUTTON_CLICK,
         isFromUserAction: true,
       });
       // Redirect user using the parent window.
-      self.open(authorizationUrl, '_parent');
+      self.setTimeout(() => {
```
Comment (fikus) [inline src/utils/gaa.js:981]: Can you file a bug and add a TODO to clean this up?

**Review by fikus — APPROVED** [review]

File: src/utils/gaa.js (hunk 2 above)
Comment (ksebina) [reply src/utils/gaa.js:982]: Yes! b/242998655

**Pull request merged**